src/plugins/discover/public/application/helpers/get_switch_index_pattern_app_state.ts

```diff
@@ -36,23 +36,38 @@ import { IndexPattern } from '../../opensearch_dashboards_services';
  * Helper function to remove or adapt the currently selected columns/sort to be valid with the next
  * index pattern, returns a new state object
  */
+
+export interface PointInTime {
+  attributes: any;
+  id: string;
+}
 export function getSwitchIndexPatternAppState(
-  currentIndexPattern: IndexPattern,
-  nextIndexPattern: IndexPattern,
+  currentIndexPattern: IndexPattern | PointInTime,
```

Comment:
In the main implementation the aim is to have the discover page react to IndexPattern and Point in Time similarly. So Here we are thinking to abstract IndexPattern and Point in Time object and have a parent interface which can be used.

Reply:
Need a separate field for point in time instead of reusing "indexpattern" fields